### ISSUE TITLE
Service not getting rule on port 8080, only on 80

### ISSUE THREAD

**Issue by aroundthecode** [istio/istio#26107]
Hi there

I'm trying to upgrade from 1.0.5 to 1.2.10 (latest working on my 1.12 K8 cluster)

I configured all My DestinationRule/Virtualservice and I notice that traffic management isn't working unless i move my k8 services from port 8080 to 80.

here is my configuration 

 ```
kind: DestinationRule
apiVersion: networking.istio.io/v1alpha3
metadata:
  name: myservice-service-destination
  namespace: flcloudns
  ...
  labels:
    app: myservice-service
    buildNum: '189'
    chart: fl-mesh-1.0.0
    heritage: Tiller
    profile: ns
    release: myservice-service
spec:
  host: myservice.flcloudns.svc.cluster.local
  trafficPolicy: ~
  subsets:
    - labels:
        version: blue
      name: blue-myservice
    - labels:
        version: green
      name: green-myservice
```

```
kind: VirtualService
apiVersion: networking.istio.io/v1alpha3
metadata:
  name: myservice-service-vservice
  namespace: flcloudns
  labels:
    app: myservice-service
    heritage: Tiller
    profile: ns
    release: myservice-service
spec:
  hosts:
    - myservice.flcloudns.svc.cluster.local
  gateways: ~
  http:
    - match:
        - headers:
            myservice:
              exact: green
      route:
        - destination:
            host: myservice.flcloudns.svc.cluster.local
            subset: green-myservice
    - match:
        - headers:
            myservice:
              exact: blue
      route:
        - destination:
            host: myservice.flcloudns.svc.cluster.local
            subset: blue-myservice
    - route:
        - destination:
            host: myservice.flcloudns.svc.cluster.local
            subset: blue-myservice
  tcp: ~
  tls: ~
  exportTo: ~
 ```
```
metadata:
  name: myservice
  namespace: flcloudns
    app: myservice
    buildNum: '189'
    chart: fl-mesh-1.0.0
    heritage: Tiller
    profile: ns
    release: myservice-service
spec:
  ports:
    - name: http-web
      protocol: TCP
      port: 8080
      targetPort: http-web
  selector:
    app: myservice
    profile: ns
  clusterIP: *********
  type: ClusterIP
  sessionAffinity: None
status:
  loadBalancer: {}
kind: Service
```

What am I supposed to do to have it work on port 8080? Just changing the port tom 8080 to 80 let traffig routes back to work.
No evidence of error in log nor kiali UI.

**Comment by aroundthecode** [comment]
I finally found the issue.
I got a non-istio headless service sitting on the same port (8080) as my istio services.

The very big problem is that route evaluation seems not to sort available routes, so the problem was NEVER SHOWING on dev environment (with the very same headless service) but ALWAYS SHOWING in test environment with no difference at all on configuration or other environment setting.

It took me months to dig out the issue (i'm struggling since istio 1.2.5! )
Is there a way to notify that a headless service is entering the mesh ?

Moreover I think that not-sorted rule evaluation could drive people crazy with troubleshooting, is it possible to have a more deterministic evaluation approach?

To anybody reaching this ticket in despair like me, to dig out the problem I had to retrieve all routing information from sidecars as descriveb here: https://istio.io/v1.2/docs/ops/traffic-management/proxy-cmd/
